src/ranking.jl

```diff
@@ -171,3 +171,170 @@ Missing values are assigned rank `missing`.
+definitions of the `quantilerank` are obtained by changing the `method` keyword argument: 
+
+`:inc` (default) - It calculates a value in the range 0 to 1 inclusive. 
+If `value ∈ v`, it returns `count_less / (n - 1)`, if not, apply interpolation based in 
+def. 7 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK` and `PERCENTRANK.INC`)
+
+`:exc` - It calculates a value in the range 0 to 1 exclusive. 
+If `value ∈ v`, it returns `(count_less + 1) / (n + 1)`,  if not, apply interpolation 
+based in def. 6 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK.EXC`)
+
+`:compete` - Based on the `competerank` of StatsBase, if `value ∈ v`, it returns 
+`count_less/ (n - 1)`, if not, returns `(count_less - 1) / (n - 1)`.
+Also, there is no interpolation.
+(equivalent to MariaDB `PERCENT_RANK`, dplyr `percent_rank`)
+
+`:tied` - Based in the def. in Roscoe, J. T. (1975), it returns 
+`(count_less + count_equal/2) / n` and there is no interpolation. 
+(equivalent to `mean` argument of Scipy `percentileofscore`)
+
+`:strict` - It returns `count_less / n` and there is no interpolation.
+(equivalent to `strict` method of Scipy `percentileofscore`)
+
+`:weak` - It returns `(count_less + count_equal) / n` and there is no interpolation.
+(equivalent to `weak` method of Scipy `percentileofscore`)
+
+!!! note
+    An `ArgumentError` is thrown if `v` contains `NaN` or `missing` values
+    and if `v` is empty or contains only one element.
+
+The function also supports types that have the mathematical properties of rings or total 
+order.
+
+# References
+[Percentile Rank on Wikipedia](https://en.wikipedia.org/wiki/Percentile_rank) covers 
+definitions and examples.
+
+Roscoe, J. T. (1975). "[Fundamental Research Statistics for the Behavioral Sciences 
+(2nd ed.)](http://www.bryanburnham.net/wp-content/uploads/2014/07/Fundamental-Statistics-
+for-the-Behavioral-Sciences-v2.0.pdf#page=57)".
+    ISBN 0-03-091934-7.
+
+Hyndman, R.J and Fan, Y. (1996) "[Sample Quantiles in Statistical Packages]
+(https://www.amherst.edu/media/view/129116/original/Sample+Quantiles.pdf)",
+    *The American Statistician*, Vol. 50, No. 4, pp. 361-365
+
+[Quantile on Wikipedia](https://en.m.wikipedia.org/wiki/Quantile) details the different 
+quantile definitions.
+
+# Examples
+```julia
+julia> using StatsBase
+
+julia> v1 = [1, 1, 1, 2, 3, 4, 8, 11, 12, 13];
+
+julia> v2 = [1, 2, 3, 5, 6, missing, 8];
+
+julia> v3 = [1, 2, 3, 4, 4, 5, 6, 7, 8, 9];
+
+# use `skipmissing` in vectors with missing entries.
+julia> quantilerank(v1, 2), quantilerank(skipmissing(v2), 4)
+(0.3333333333333333, 0.5)
+```
+
+# use `Ref` to treat vector `v3` as a scalar during broadcasting.
+```julia
+julia> quantilerank.(Ref(v3), [4, 8])
+2-element Vector{Float64}:
+ 0.3333333333333333
+ 0.8888888888888888
+```
+"""
+function quantilerank(v, value; method::Symbol=:inc)
+    # checks
+    value isa Number && isnan(value) &&
```

Comment:
why don't we check that `value` is `missing`?

Reply:
Done in commit 9b4d2aa